include/openssl/base.h

```diff
@@ -349,6 +349,13 @@ typedef struct md4_state_st MD4_CTX;
 typedef struct md5_state_st MD5_CTX;
 typedef struct ocsp_req_ctx_st OCSP_REQ_CTX;
 typedef struct ossl_init_settings_st OPENSSL_INIT_SETTINGS;
+typedef struct pkcs7_digest_st PKCS7_DIGEST;
+typedef struct pkcs7_enc_content_st PKCS7_ENC_CONTENT;
+typedef struct pkcs7_encrypt_st PKCS7_ENCRYPT;
+typedef struct pkcs7_envelope_st PKCS7_ENVELOPE;
+typedef struct pkcs7_issuer_and_serial_st PKCS7_ISSUER_AND_SERIAL;
```

Comment:
`PKCS7_ISSUER_AND_SERIAL` seems to only be nested internally within the opaque `PKCS7` structures. Maybe we could move the `typedef` to `internal.h` and not expose this externally?

Note: Did a search and Ruby doesn't consume this directly. A lot of these are intertwined with non-opaque structures, so looks like we'll have to expose most.

Reply:
yep, agreed.